infra/main.tf

```
@@ -39,6 +39,16 @@ resource "fastly_service_vcl" "peps" {
     EOT
   }
 
+  snippet {
+    name    = "topics"
+    type    = "recv"
+    content = <<-EOT
+        if (req.url == "/topics") {
+          set req.url = "/topic/";
```

At least judging from the others and skimming the Fastly docs, right now, this not only isn't matching `/`, but also not matching or including URLs that actually specify a topic either, and nor is it performing a redirect (as it needs to set `req.http.location` rather than `req.url`, and then error with the right error code to pass control to the redirect handler.

Following the others, you could do

```suggestion
        if (req.url ~ "^/topics(?:/([\w-]*/?))?$") {
          set req.http.Location = "/topics/" + re.group.1;
          error 618;
```

However, instead of capturing and then replacing, seems like it might be simpler and more robust to just do

```suggestion
        if (req.url ~ "^/topics($|/)") {
          set req.http.Location = regsub(req.http.Location, "^/topics/?", "/topic/");
          error 618;
```